### PR TITLE
Add Spark Masthead And Create Homepage

### LINKED ISSUE #2298: Build Out Homepage - NDS
**User Story:**
As Spark we want to build out a homepage for for our New Doc Site.

**Notes:**
- Work with Jane
- See Invision link for wireframe Jane Created

**AC:**
NEED TO SEE NEW INVISION LINK

### PR DESCRIPTION
**Pull request by afebbraro**
## What does this PR do?
Updates the Gatsby site to use the Spark Masthead. Adds in custom react static query hooks to fetch all the page data at build time to generate the list of links in the Spark Masthead. Removes unneeded SCSS (now that we are using spark). Changes the page layout of the Gatsby site.

* waiting on content and images for teaser cards

### Associated Issue 
Fixes #2298 and #2627

## Please check off completed items as you work.
If a checklist item or section does not apply to your PR
then please remove it.

### Code
 - [x] Build Component in Spark React
 - [x] Unit Testing in Spark React with `gulp test-react` (100% coverage, 100% passing)

### Accessibility
- [x] New changes abide by [accessibility requirements](https://sparkdesignsystem.com/docs/accessibility)

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome
  - [ ] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [ ] Mozilla Firefox (Mobile)
  - [ ] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [ ] Microsoft Edge
  - [ ] Apple Safari
  - [ ] Apple Safari (Mobile) -- noticed an issue with masthead here and am working through it
  
### Design Review
 - [ ] Design reviewed and approved

### CODE REVIEW

File: react/src/components/card/SprkCard.test.js [hunk 1]
```diff
@@ -60,6 +60,36 @@ describe('SprkCard:', () => {
     expect(loadsTeaserComponent).toBe(true);
   });
 
+  it('should apply the correct button variant to cta button in teaser card', () => {
```
Comment (afebbraro) [inline react/src/components/card/SprkCard.test.js]: This adds a test for the new variant.

File: src/components/Menu.js [hunk 2]
```diff
@@ -1,21 +1,13 @@
 import React from 'react';
 import propTypes from 'prop-types';
+import { SprkDivider } from '@sparkdesignsystem/spark-react';
```
Comment (bobwise) [inline src/components/Menu.js]: can be removed

File: gatsby-browser.js [hunk 3]
```diff
@@ -9,7 +9,18 @@ exports.onRouteUpdate = () => {
       apiKey: '9486baf438a1d7fd0b6c982838d6f9bc',
       indexName: 'sparkdesignsystem',
       inputSelector: '.docs-header-search input',
-      debug: 'false'
+      debug: 'false',
+    });
+
+    docsearch({ // eslint-disable-line no-undef
+      apiKey: '9486baf438a1d7fd0b6c982838d6f9bc',
```
[the remark this answers is not in the document]
Comment (afebbraro) [reply gatsby-browser.js]: @bobwise, I asked Rob that question as he was the one who originally put it in there. This is duplicating what he had put in there. I talked to @yodasw16  about this and he is going to sync back up with @RCopeland about it.

File: src/scss/components/_header.scss [hunk 4]
```diff
@@ -1,37 +1,21 @@
...
+  flex: 0 0 auto;
 }
 
-.docs-header__menu {
-  @media (max-width: $doc-site-mobile-breakpoint) {
-    display: none;
-  }
-}
+.docs-masthead .sprk-c-Masthead__branding {
+  padding-right: 48px;
```
[the remark this answers is not in the document]
Comment (afebbraro) [reply src/scss/components/_header.scss]: good question, the spark masthead expects a "sign in" type link on the right but for our docs site we dont have a link on the right. So, this is a one-off customization for us. I'm using the "sign in" space for our docs site search input but dropped it to the next line so users have more space for the input.

File: src/components/ContextMenu.js [hunk 5]
```diff
@@ -2,38 +2,87 @@ import React from 'react';
 import { SprkLink } from '@sparkdesignsystem/spark-react';
 import { Link } from 'gatsby';
 import classnames from 'classnames';
+import PropTypes from 'prop-types';
```
[the remark this answers is not in the document]
Comment (afebbraro) [reply src/components/ContextMenu.js]: You're right!! Great catch! Removing now.

File: gatsby-browser.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (RCopeland) [reply gatsby-browser.js]: yep, thats actually a public safe API key we get from algolia

[the remark this answers is not in the document]
Comment (RCopeland) [reply gatsby-browser.js]: its meant for static sites and is fine to commit

**Pull request merged**